rules/TypeDeclaration/Helper/PhpDocNullableTypeHelper.php

```diff
@@ -133,29 +134,38 @@ private function resolveUpdatedPhpDocTypeFromPhpDocTypeAndPhpParserTypeNullInfo(
     ): ?Type {
         /** @var array<(NullType | UnionType)> $updatedDocTypes */
         $updatedDocTypes = [];
-        $phpDocTypeContainsNullType = false;
+        $isPhpDocTypeContainingNullType = false;
+        $isPhpDocTypeContainingClosureType = $phpDocType instanceof ClosureType;
         if ($phpDocType instanceof UnionType) {
-            $phpDocTypeContainsNullType = TypeCombinator::containsNull($phpDocType);
+            $isPhpDocTypeContainingNullType = TypeCombinator::containsNull($phpDocType);
             foreach ($phpDocType->getTypes() as $subType) {
                 if ($subType instanceof NullType) {
                     continue;
                 }
 
+                if ($subType instanceof ClosureType) {
+                    $isPhpDocTypeContainingClosureType = true;
+                }
+
                 $updatedDocTypes[] = $subType;
             }
         } else {
             $updatedDocTypes[] = $phpDocType;
         }
 
         if (! $this->isItRequiredToRemoveOrAddNullTypeToUnion(
-            $phpDocTypeContainsNullType,
+            $isPhpDocTypeContainingNullType,
             $isPhpParserTypeContainingNullType
         )) {
             return null;
         }
 
         if ($isPhpParserTypeContainingNullType) {
-            $updatedDocTypes[] = new NullType();
+            if ($isPhpDocTypeContainingClosureType) {
+                array_unshift($updatedDocTypes, new NullType());
+            } else {
+                $updatedDocTypes[] = new NullType();
+            }
```

Comment:
you can create separate private method for this, return updated `$pudatedDocTypes`